packages/components/src/renderer/FileMessage/index.tsx

```diff
@@ -1,53 +1,52 @@
 import mimeTypes from 'mime/lite'
-import React from 'react'
 import path from 'path'
+import React, { useMemo } from 'react'
 import { MessageTypeHandlerProps } from '../../typings'
 
-export const File = ({ payload }: MessageTypeHandlerProps<'file' | 'video' | 'audio'>) => {
-  if (!payload) {
-    return null
-  }
-
-  const { url, title } = payload
-
-  let extension = ''
-  try {
-    const validUrl = new URL(url)
-
-    extension = validUrl.pathname
-  } catch (error) {
-    // Try using path.extname since url might be relative.
-    extension = path.extname(url)
-  }
-
-  const mime = mimeTypes.getType(extension)
-
-  if (mime?.includes('image/')) {
-    return (
-      <a href={url} target={'_blank'}>
-        <img src={url} title={title} />
-      </a>
-    )
-  } else if (mime?.includes('audio/')) {
-    return (
+export const Audio: React.FC<MessageTypeHandlerProps<'audio'>> = ({ audio, title }) => {
+  const mime = useMimeType(audio)
+  return (
+    <div>
+      <span>{title}</span>
       <audio controls>
-        <source src={url} type={mime} />
+        <source src={audio} type={mime} />
       </audio>
-    )
-  } else if (mime?.includes('video/')) {
-    return (
-      <video controls>
-        <source src={url} type={mime} />
-      </video>
-    )
-  }
+    </div>
+  )
+}
 
+export const File: React.FC<MessageTypeHandlerProps<'file'>> = ({ file, title }) => {
   return (
     <div>
       <span>File: </span>
-      <a href={url} target={'_blank'}>
-        {title || url}
+      <a href={file} target={'_blank'}>
```

Comment:
This should remove the warning.

```suggestion
      <a href={file} target={'_blank'} rel="noopener noreferrer">
```